From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andy Dill <andy.dill@gmail.com>
Date: Fri, 14 Feb 2020 12:26:51 -0800
Subject: fix: normalize excessively long paths in blink bindings generator

Combining paths in this manner without normalization makes it very,
very easy to butt up against MAX_PATH on Windows.

diff --git a/third_party/blink/renderer/bindings/scripts/bind_gen/__init__.py b/third_party/blink/renderer/bindings/scripts/bind_gen/__init__.py
index 048f3b7eb8a4..faf40a8008cb 100644
--- a/third_party/blink/renderer/bindings/scripts/bind_gen/__init__.py
+++ b/third_party/blink/renderer/bindings/scripts/bind_gen/__init__.py
@@ -12,7 +12,8 @@ def _setup_sys_path():
     expected_path = 'third_party/blink/renderer/bindings/scripts/bind_gen/'

     this_dir = os.path.dirname(__file__)
-    root_dir = os.path.join(this_dir, *(['..'] * expected_path.count('/')))
+    root_dir = os.path.normpath(
+        os.path.join(this_dir, *(['..'] * expected_path.count('/'))))

     sys.path = [
         # //third_party/blink/renderer/bindings/scripts/web_idl
diff --git a/third_party/blink/renderer/bindings/scripts/web_idl/__init__.py b/third_party/blink/renderer/bindings/scripts/web_idl/__init__.py
index a2fb2b26bbaf..6b146fa92853 100644
--- a/third_party/blink/renderer/bindings/scripts/web_idl/__init__.py
+++ b/third_party/blink/renderer/bindings/scripts/web_idl/__init__.py
@@ -12,7 +12,8 @@ def _setup_sys_path():
     expected_path = 'third_party/blink/renderer/bindings/scripts/web_idl/'

     this_dir = os.path.dirname(__file__)
-    root_dir = os.path.join(this_dir, *(['..'] * expected_path.count('/')))
+    root_dir = os.path.normpath(
+        os.path.join(this_dir, *(['..'] * expected_path.count('/'))))
     sys.path = [
         # //third_party/blink/renderer/build/scripts/blinkbuild
         os.path.join(root_dir, 'third_party', 'blink', 'renderer', 'build',
